core/services/ocr3/plugins/ccip/internal/reader/home_chain.go

```diff
@@ -0,0 +1,285 @@
+		// empty set to denote no chains supported
+		return mapset.NewSet[cciptypes.ChainSelector](), nil
+	}
+	return s.nodeSupportedChains[id], nil
+}
+
+func (r *HomeChainConfigPoller) GetKnownCCIPChains() (mapset.Set[cciptypes.ChainSelector], error) {
+	r.mutex.RLock()
+	defer r.mutex.RUnlock()
+	knownSourceChains := mapset.NewSet[cciptypes.ChainSelector]()
+	for chain := range r.state.chainConfigs {
+		knownSourceChains.Add(chain)
+	}
+
+	return knownSourceChains, nil
+}
+
+func (r *HomeChainConfigPoller) GetFChain() (map[cciptypes.ChainSelector]int, error) {
+	r.mutex.RLock()
+	defer r.mutex.RUnlock()
+	return r.state.fChain, nil
+}
+
+func (r *HomeChainConfigPoller) Close() error {
+	return r.sync.StopOnce(r.Name(), func() error {
+		close(r.stopCh)
+		return nil
+	})
+}
+
+func (r *HomeChainConfigPoller) Ready() error {
+	r.mutex.RLock()
+	defer r.mutex.RUnlock()
+	return r.sync.Ready()
+}
+
+func (r *HomeChainConfigPoller) HealthReport() map[string]error {
+	r.mutex.RLock()
+	defer r.mutex.RUnlock()
+	if r.failedPolls >= MaxFailedPolls {
+		r.sync.SvcErrBuffer.Append(fmt.Errorf("polling failed %d times in a row", MaxFailedPolls))
+	}
+	return map[string]error{r.Name(): r.sync.Healthy()}
+}
+
+func (r *HomeChainConfigPoller) Name() string {
+	return "HomeChainConfigPoller"
+}
+
+func createFChain(chainConfigs map[cciptypes.ChainSelector]ChainConfig) map[cciptypes.ChainSelector]int {
+	fChain := map[cciptypes.ChainSelector]int{}
+	for chain, config := range chainConfigs {
+		fChain[chain] = config.FChain
+	}
+	return fChain
+}
+
+func createKnownChains(chainConfigs map[cciptypes.ChainSelector]ChainConfig) mapset.Set[cciptypes.ChainSelector] {
+	knownChains := mapset.NewSet[cciptypes.ChainSelector]()
+	for chain := range chainConfigs {
+		knownChains.Add(chain)
+	}
+	return knownChains
+}
+
+func createNodesSupportedChains(chainConfigs map[cciptypes.ChainSelector]ChainConfig) map[libocrtypes.PeerID]mapset.Set[cciptypes.ChainSelector] {
+	nodeSupportedChains := map[libocrtypes.PeerID]mapset.Set[cciptypes.ChainSelector]{}
+	for chainSelector, config := range chainConfigs {
+		for _, p2pID := range config.SupportedNodes.ToSlice() {
+			if _, ok := nodeSupportedChains[p2pID]; !ok {
+				nodeSupportedChains[p2pID] = mapset.NewSet[cciptypes.ChainSelector]()
+			}
+			//add chain to SupportedChains
+			nodeSupportedChains[p2pID].Add(chainSelector)
+		}
+	}
+	return nodeSupportedChains
+}
+
+func (r *HomeChainConfigPoller) fetchOnChainConfig(ctx context.Context) ([]ChainConfigInfo, error) {
+	r.lggr.Infow("Fetching HomeChainConfigMapper")
+	var chainConfigInfo []ChainConfigInfo
+	err := r.homeChainReader.GetLatestValue(ctx, "CCIPCapabilityConfiguration", "getAllChainConfigs", nil, &chainConfigInfo)
+	if err != nil {
+		return nil, err
+	}
+	return chainConfigInfo, err
+}
+
+func (r *HomeChainConfigPoller) convertOnChainConfigToHomeChainConfig(capabilityConfigs []ChainConfigInfo) (map[cciptypes.ChainSelector]ChainConfig, error) {
```

Comment:
This appears to be a pure function
```suggestion
func convertOnChainConfigToHomeChainConfig(capabilityConfigs []ChainConfigInfo) (map[cciptypes.ChainSelector]ChainConfig, error) {
```